### ISSUE TITLE
Need help with config

### ISSUE THREAD

**Issue by Semih67** [Unity-Technologies/ml-agents#1690]
So I'm programming a game where the has to shoot 3 opponents. The opponents have also the ability to shoot. They get negative reward if they are to far away to the player and positive if they are closer. And if they looking at the player and shoot at him they get positive reward. But I'm always dividing the reward with the maxSteps: `AddReward(0.75f / agentParameters.maxStep);` 
Maybe this is wrong? So my Graphs look like this: 
![entropy](https://user-images.githubusercontent.com/33628476/52463915-167f7600-2b79-11e9-9caf-06845c99e62b.PNG)
![reward](https://user-images.githubusercontent.com/33628476/52463916-18493980-2b79-11e9-952a-f396b63ac5fd.PNG)
![all](https://user-images.githubusercontent.com/33628476/52463918-1a12fd00-2b79-11e9-97fd-c46e85d96f85.PNG)

The entropy is not decreasing while I'm at the 4millth step. Always just for a moment and increases again. And the rewards are going up and down. This is my config.yaml: 

```
trainer: ppo
    batch_size: 80
    beta: 3.0e-3
    buffer_size: 15240
    epsilon: 0.2
    gamma: 0.86
    hidden_units: 75
    lambd: 0.925
    learning_rate: 9.0e-3
    max_steps: 5.0e6
    memory_size: 256
    normalize: false
    num_epoch: 5
    num_layers: 1
    time_horizon: 100
    sequence_length: 64
    summary_freq: 1000
    use_recurrent: false
    use_curiosity: false
    curiosity_strength: 0.01
    curiosity_enc_size: 128
```

I have only one hidden layer because I thought this are simple tasks. The opponent has to move, look at the player and shoot at him. Maybe I need more then one layer.  And my agents brain is discrete. The opponents are also barely moving around. It took for one of them 3 mio steps to move and the other ones still do not move. So how would you solve this? Which parameter of the config would you change? And also here are my observations: 

```
AddVectorObs(rayPer.Perceive(rayDistance, rayAngles, detectableObjects, 0f, 0f));
            AddVectorObs(rayPer.Perceive(rayDistance, rayAngles, detectableObjects, 1.5f, 0f));
            AddVectorObs(academy.player.transform.position.x);
            AddVectorObs(academy.player.transform.position.z);
            AddVectorObs(gameObject.transform.position.x);
            AddVectorObs(gameObject.transform.position.z);
            AddVectorObs(gameObject.transform.rotation.y);
            AddVectorObs(distanceOfPlayer);
            AddVectorObs(hitDistanceGlobal);
            AddVectorObs(academy.player.transform.position.z - gameObject.transform.position.z);
            AddVectorObs(academy.player.transform.position.x - gameObject.transform.position.x);
```
Don't know if I'm using the correct vectors to observe the environment.

So I need your help

Thanks in advance

**Comment by MarcoMeter** [comment]
Hi @Semih67 

the first issue that I can identify is a missing normalization of observation space.
You can normalize your observations manually as described in https://github.com/Unity-Technologies/ml-agents/blob/master/docs/Learning-Environment-Best-Practices.md
or you can set normalize to true in your config file.

Your reward function sounds pretty weird. If you divide 0.75 by maxStep, which is 5 million in your case, you pretty much signal a reward of value 0.

Further, it would be interesting to know the setup of your action space. Is it continuous or is it discrete? What are the actions?

Your observation space comprises distances and positions. A velocity would be nice to add. Also, relative positions to the agent could be an idea as well. Then you could actually omit the distance.


At last I quite don't understand who the agent is. If agents', working with the same brains, are supposed to compete each other, this adds lots of complexity to the to be trained problem.

**Comment by Semih67** [comment]
The actions are discrete. And the agents are playing against the player. They have one brain and the target is the same. So I should delete division and add a normalization? and what about the hidden layer? Is it fine to set it to 1 or should I increase it to 2?

**Comment by ervteng** [comment]
I'd leave the hidden layer as is for now. As for the reward, as long as the reward at each timestep is under 1 (your example is 0.75), there isn't much of a reason to be dividing by a huge number such as maxSteps. 

As @MarcoMeter said, the easiest way to normalize will be to turn on normalization in the config. You also have a lot of redundant information in the observation - for instance, both the distance as well as the vector to the player. These shouldn't affect the ability for the agent to learn, but it may slow the learning process as the agent has to deal with a larger observation space. 

The other missing variable is, what is the player doing during training? Is it randomly moving around?

**Comment by Semih67** [comment]
yes during the training process the player is randomly moving around. And if its finished one can controll the player.  So I can add the velocity of the player to the observation and what else?

**Comment by ervteng** [comment]
That is odd. Can you post the reward and loss curves? 

https://github.com/Unity-Technologies/ml-agents/blob/master/docs/Using-Tensorboard.md

Do the agents move occasionally, or are they completely dead?

**Comment by ervteng** [comment]
Closing this issue due to inactivity. Feel free to post a new issue if you're still having issues. Thanks!

**Comment by Semih67** [comment]
Hey @ervtang I have opened a new issue of my game but I did not get any answer: https://github.com/Unity-Technologies/ml-agents/issues/1805 

I hope you can look at this

This describes my problem in details
